File: tests/testdata/codegen/depends-on-array-pp/nodejs/depends-on-array.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const myBucket = new aws.s3.Bucket("myBucket", {website: {
    indexDocument: "index.html",
}});
const ownershipControls = new aws.s3.BucketOwnershipControls("ownershipControls", {
    bucket: myBucket.id,
    rule: {
        objectOwnership: "ObjectWriter",
    },
});
const publicAccessBlock = new aws.s3.BucketPublicAccessBlock("publicAccessBlock", {
    bucket: myBucket.id,
    blockPublicAcls: false,
});
const indexHtml = new aws.s3.BucketObject("index.html", {
    bucket: myBucket.id,
    source: new pulumi.asset.FileAsset("./index.html"),
    contentType: "text/html",
    acl: "public-read",
}, {
    dependsOn: [
        publicAccessBlock,
        ownershipControls,
    ],
});
export const bucketName = myBucket.id;
export const bucketEndpoint = pulumi.interpolate`http://${myBucket.websiteEndpoint}`;

File: tests/testdata/codegen/depends-on-array-pp/python/depends-on-array.py
import pulumi
import pulumi_aws as aws

my_bucket = aws.s3.Bucket("myBucket", website={
    "index_document": "index.html",
})
ownership_controls = aws.s3.BucketOwnershipControls("ownershipControls",
    bucket=my_bucket.id,
    rule={
        "object_ownership": "ObjectWriter",
    })
public_access_block = aws.s3.BucketPublicAccessBlock("publicAccessBlock",
    bucket=my_bucket.id,
    block_public_acls=False)
index_html = aws.s3.BucketObject("index.html",
    bucket=my_bucket.id,
    source=pulumi.FileAsset("./index.html"),
    content_type="text/html",
    acl="public-read",
    opts = pulumi.ResourceOptions(depends_on=[
            public_access_block,
            ownership_controls,
        ]))
pulumi.export("bucketName", my_bucket.id)
pulumi.export("bucketEndpoint", my_bucket.website_endpoint.apply(lambda website_endpoint: f"http://{website_endpoint}"))
